README.md

```diff
@@ -1,9 +1,106 @@
-This is a [demo app](https://web.dev/porting-libusb-to-webusb/) running gPhoto2 on the Web:
+# Web-gPhoto2
 
-![A picture of DSLR camera connected via a USB cable to a laptop. The laptop is running the Web demo mentioned in the article, which mirrors a live video feed from the camera as well as allows to tweak its settings via form controls.](https://web-dev.imgix.net/image/9oK23mr86lhFOwKaoYZ4EySNFp02/MR4YGRvl0Z9AWT6vv3sQ.jpg?auto=format&w=1600)
+A gPhoto2 implementation using WebAssembly to control DSLR cameras from the browser.
 
 Powered by a [custom fork](https://github.com/RReverser/libgphoto2) of [libgphoto2](https://github.com/gphoto/libgphoto2), the [WebUSB](https://github.com/WICG/webusb) backend of [libusb](https://github.com/libusb/libusb), and WebAssembly via [Emscripten](https://emscripten.org/).
 
+# NPM
+
+## Installation
+
+```bash
+npm install web-gphoto2
+// or
+yarn add web-gphoto2
+```
+
+## Usage
+
+A short example on how to use this package:
+
+```ts
+import Camera from "web-gphoto2";
+
+let camera = new Camera();
+
+async function connectCamera() {
+  await camera.showCameraPicker();
+  await camera.connect();
+}
+
+async function getSupportedOps() {
+  const ops = await camera.getSupportedOps();
+  console.log("Supported Ops:", ops);
+}
+
+async function getCameraConfig() {
+  const config = await camera.getConfig();
+  console.log("Config:", config);
+}
+
+async function updateConfig() {
+  await camera.setConfigValue("iso", "800");
+}
+
+async function capturePreviewAsBlob() {
+  const blob = await camera.capturePreviewAsBlob();
+  console.log("Blob:", blob);
+}
+
+async function captureImageAsFile() {
+  const file = await camera.captureImageAsFile();
+  console.log("File:", file);
+}
+```
+
+## Common Issues
+
+### SharedArrayBuffer can not be found
+
+SharedArrayBuffer has been disabled across all browsers due to the Spectre vulnerability. This package uses SharedArrayBuffer to communicate with the WebAssembly module. To work around this issue, you need to set two response headers for your document:
+
+```
+Cross-Origin-Opener-Policy: same-origin
+Cross-Origin-Embedder-Policy: require-corp
+```
+
+Information from [Stackoverflow](https://stackoverflow.com/questions/64650119/react-error-sharedarraybuffer-is-not-defined-in-firefox)
+
+### Error: Not found: /node_modules/.vite/deps/libapi.wasm
+
+Vite tries to optimize the dependencies by default. This causes the WebAssembly module to be moved to a different location. To prevent this, you need to exclude the web-gphoto2 package from the optimization.
```

Comment:
Not sure how I feel about including framework-specific workarounds tbh, both because of noise for non-Vite users in README and because we need to maintain them when/if this changes... is there a link to docs or stackoverflow or something could point to here instead?

Reply:
I completely understand your concern but this is not a widely documented issue and it took me a while to figure out why I was getting weird errors during development. I found the information in this issue: https://github.com/vitejs/vite/issues/8497 which is a duplicate of this issue: https://github.com/vitejs/vite/issues/8427 but that issue is still open and is not one that is found easily when looking to solve the issue.

The way I see it: This short message saves some people a lot of time and only takes little time (just reading the title is enough) for those that don't need it. 

What do you think of moving the Common Issues section all the way to the bottom? That way people looking for any other information won't even see it.

Reply:
Hmm I see. Maybe just hide it in collapsed `<details>` for now, I hope we'll at least be able to remove it in the future. https://gist.github.com/scmx/eca72d44afee0113ceb0349dd54a84a2

Reply:
I moved the common issues section all the way to the bottom (just before 'See also' and 'License')